### PR TITLE
Add "Locally Euclidean -> Locally Compact"

### PR DESCRIPTION
**Pull request by ccaruvana**
This gives us a list of spaces which are locally metrizable but not locally Euclidean.

### CODE REVIEW

**Comment by ccaruvana** [comment]
That's a good question. It seems like the potential issue for showing "locally T3 -> T3" is that closed subsets of open subsets need not be closed. This could be a nice thing to flesh out and add to the list of other "locally P -> P" theorems.

On the other hand, I think locally Euclidean does imply regular by the same argument showing that it's locally compact. The compact neighborhoods are compact globally, so every point has a basis consisting of closed sets.

**Comment by ccaruvana** [comment]
Actually, having said that, shouldn't "locally compact -> regular" be added instead? At least with Hausdorff, we have https://math.stackexchange.com/questions/1313688. I guess the necessary component for the argument I said above is Hausdorffness since compact subsets of Hausdorff spaces are closed. So maybe we should add "locally compact + Hausdorff -> regular". We have recently talked about locally T2 -> T2, so adding this would show that locally Euclidean is locally T2 so then T2 and locally compact. Mainly, I would only suggest this route since we already have a stackexchange reference for locally Euclidean -> locally compact, but I'm open to suggestions on how to handle all of the above.

**Comment by prabau** [comment]
The problem is that in a non-Hausdorff case a compact set need not be closed.  It seems that locally Euclidean does not imply regular after all.  The [line with two origins](https://en.wikipedia.org/wiki/Non-Hausdorff_manifold) is an example.  The two "origins" are closed points, but not separated by disjoint nbhds.  So locally T3 does not imply T3 either.  We should add that example to pi-base at some point, to get an example of non-Hausdorff locally Euclidean space.

Anyway, that example shows that in locally Euclidean space you can have a local base of compact nbhds, but these compact nbhds need to be compact closed nbhds.

**Comment by ccaruvana** [comment]
Ah, yes. Thanks. I realized also that we had "locally T1 -> T1". Thanks for clearing that up.

**Comment by prabau** [comment]
Regarding the various flavors of locally compact (https://en.wikipedia.org/wiki/Locally_compact_space):
For background info, Steven and I had a long discussion about what's best to have in issue #157.
We ended up with:
- weakly locally compact = at least one compact nbhd
- locally compact - local base of compact nbhds
- weakly locally compact + regular = local base of closed compact nbhds
- locally relatively compact = something ...

One of the most useful combinations is "local base of closed compact nbhds" = LC + regular = WLC + regular.  But LC by itself is a weaker property.  "WLC + regular" is an easier condition to check than "LC + regularl".  That's why in https://topology.pi-base.org/properties/P000023/theorems that are many theorems with that as a hypothesis.

Now regarding your comment, LC does not imply regular (lots of examples of LC + ~ regular). And [LC + T2 ==> regular ] is also already covered (as can be seen from the search form by searching for the negation of the result).

**Comment by prabau** [comment]
Apart from all that, your result here is fine.  Let me give some minor nitpicks on the file itself.

**Comment by ccaruvana** [comment]
> The problem is that in a non-Hausdorff case a compact set need not be closed. It seems that locally Euclidean does not imply regular after all. The [line with two origins](https://en.wikipedia.org/wiki/Non-Hausdorff_manifold) is an example. The two "origins" are closed points, but not separated by disjoint nbhds. So locally T3 does not imply T3 either. We should add that example to pi-base at some point, to get an example of non-Hausdorff locally Euclidean space.
> 
> Anyway, that example shows that in locally Euclidean space you can have a local base of compact nbhds, but these compact nbhds need to be compact closed nbhds.
> 
> (was writing this before having read your last comment. let me read that now)

Just to note the somewhat obvious for posterity, but this example also shows that locally T2 does not imply T2.

File: theorems/T000332.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  P000130: true
+
+refs:
+  - mathse: 103774
+    name: Every manifold is locally compact?
+---
+
+Every point has a neighborhood that is {P123}, so homeomorphic to $\mathbb R^n$ for some non-negative integer $n$. Since $\mathbb R^n$ is {P130}, it produces a basis of compact sets.
+
+See also [ProofWiki: Locally Euclidean Space is Locally Compact](https://proofwiki.org/wiki/Locally_Euclidean_Space_is_Locally_Compact).
```
Comment (prabau) [inline theorems/T000332.md:15]: I really don't like the style of ProofWiki, always belaboring the point and very hard to read in my opinion.  The mathse reference seems enough?

**Review by prabau — APPROVED** [review]

**Comment by prabau** [comment]
@ccaruvana Sorry, I messed up (but not in a big way).  I clicked on "rebase and merge", thinking it would allow to squash things.  I should have clicked on "squash and merge".  Anyway, the end result is the same, just with a slightly messy git log.  Will do better next time :-)

**Pull request merged**